modules/openxr/openxr_api.cpp

```diff
@@ -1262,10 +1276,14 @@ void OpenXRAPI::set_view_configuration(XrViewConfigurationType p_view_configurat
 	view_configuration = p_view_configuration;
 }
 
-void OpenXRAPI::set_requested_reference_space(XrReferenceSpaceType p_requested_reference_space) {
-	ERR_FAIL_COND(is_initialized());
-
+bool OpenXRAPI::set_requested_reference_space(XrReferenceSpaceType p_requested_reference_space) {
 	requested_reference_space = p_requested_reference_space;
+
+	if (is_initialized()) {
+		return setup_play_space();
```

Comment:
One thing we should add here is calling into the XRServer and resetting our reference frame transform. A user shouldn't be using `center_on_hmd` in combination with `local` and `local floor` reference spaces but just in case they did...

Similarly we should probably do that as well when the user trigger a recenter just before we emit the pose recenter signal, but only if the reference space isn't `stage`.

Reply:
> One thing we should add here is calling into the XRServer and resetting our reference frame transform. A user shouldn't be using `center_on_hmd` in combination with `local` and `local floor` reference spaces but just in case they did...

Ah, good point! If they were in `STAGE`, used `center_on_hmd()`, then switched to `LOCAL` or `LOCAL_FLOOR`, they'd be stuck with the old reference frame.

In my latest push I've added an `XRServer::clear_reference_frame()` function which is called when changing the reference space. I've also exposed it, because I could imagine scenarios where a developer would want to clear any previous recentering, for example, when switching from VR to MR mode.

> Similarly we should probably do that as well when the user trigger a recenter just before we emit the pose recenter signal, but only if the reference space isn't `stage`.

I'm less sure about this one. If we clear it when switching to `LOCAL` or `LOCAL_FLOOR`, and we give the developer the ability to clear it when they need to, I'm not sure this is necessary.